wgsl/index.bs

```diff
@@ -2747,6 +2747,34 @@ TODO(dsinclair): Add Level-of-Detail via explicit gradient. "Grad" image operand
 TODO(dsinclair): Need gather operations
 </pre>
 
+## Subgroup built-in functions ## {#subgroup-builtin-functions}
+
+<table class='data'>
+  <thead>
+    <tr><td>Subgroup built-in functions<td>SPIR-V
+  </thead>
+  <tr><td>subgroup_size() -&gt; u32<td>SubgroupSize
+  <tr><td>subgroup_local_index() -&gt; u32<td>SubgroupLocalInvocationId
+  <tr><td>subgroup_is_first() -&gt; bool<td>OpGroupNonUniformElect
+  <tr><td>subgroup_all(bool) -&gt; bool<td>OpGroupNonUniformAll
+  <tr><td>subgroup_any(bool) -&gt; bool<td>OpGroupNonUniformAny
+  <tr><td>subgroup_ballot(bool) -&gt; vec2&lt;u32&gt;<td>OpGroupNonUniformBallot
+  <tr><td>subgroup_broadcast(*T*,u32) -&gt; *T*<td>OpGroupNonUniformBroadcast
+  <tr><td>subgroup_broadcast_first(*T*) -&gt; *T*<td>OpGroupNonUniformBroadcastFirst
+  <tr><td>subgroup_add(*T*) -&gt; *T*<td>OpGroupNonUniformIAdd or OpGroupNonUniformFAdd with Reduce
+  <tr><td>subgroup_mul(*T*) -&gt; *T*<td>OpGroupNonUniformIMul or OpGroupNonUniformFMul with Reduce
+  <tr><td>subgroup_min(*T*) -&gt; *T*<td>OpGroupNonUniformUMin or OpGroupNonUniformSMin or OpGroupNonUniformFMin with Reduce
+  <tr><td>subgroup_max(*T*) -&gt; *T*<td>OpGroupNonUniformUMax or OpGroupNonUniformSMax or OpGroupNonUniformFMax with Reduce
+  <tr><td>subgroup_and(*T*) -&gt; *T*<td>OpGroupNonUniformBitwiseAnd
+  <tr><td>subgroup_or(*T*) -&gt; *T*<td>OpGroupNonUniformBitwiseOr
+  <tr><td>subgroup_xor(*T*) -&gt; *T*<td>OpGroupNonUniformBitwiseXor
+  <tr><td>subgroup_prefix_add(*T*) -&gt; *T*<td>OpGroupNonUniformIAdd or OpGroupNonUniformFAdd with ExclusiveScan
+  <tr><td>subgroup_prefix_mul(*T*) -&gt; *T*<td>OpGroupNonUniformIMul or OpGroupNonUniformFMul with ExclusiveScan
```

Comment:
For all these where the param is *T*, what are the possible values of T?

Reply:
I have tried to explicitly state the possible types, does it look OK now?